source/GameInterface/Services/Settlements/Patches/ChangeOwnerOfSettlementPatch.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Services.Settlements.Messages;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.Library.NewsManager;
+using static TaleWorlds.CampaignSystem.Actions.ChangeOwnerOfSettlementAction;
+
+namespace GameInterface.Services.Settlements.Patches
+{
+    /// <summary>
+    /// Patches ownership changes of settlements
+    /// </summary>
+    [HarmonyPatch(typeof(ChangeOwnerOfSettlementAction), "ApplyInternal")]
+    public class ChangeOwnerOfSettlementPatch
+    {
+        private static readonly AllowedInstance<Settlement> AllowedInstance = new AllowedInstance<Settlement>();
+        private static readonly MethodInfo _applyInternal = typeof(ChangeOwnerOfSettlementAction).GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
This is apparently much faster than reflection. Try to use this from now on.

The syntax is a bit gross but I think it's worth learning

```suggestion
        private static readonly Action<Settlement, Hero, Hero, ChangeOwnerOfSettlementAction.ChangeOwnerOfSettlementDetail> ApplyInternal = 
        typeof(ChangeOwnerOfSettlementAction)
        .GetMethod("ApplyInternal", BindingFlags.NonPublic | BindingFlags.Static)
        .BuildDelegate<Action<Settlement, Hero, Hero, ChangeOwnerOfSettlementAction.ChangeOwnerOfSettlementDetail>>();
```